test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -36,6 +39,298 @@ public class HubEndpointTests
+
+                var invocationAdapter = serviceProvider.GetService<InvocationAdapterRegistry>();
+                var writer = invocationAdapter.GetInvocationAdapter("json");
+                var serializer = new JsonSerializer();
+
+                await SendRequest(firstConnection.HttpConnection, writer, "BroadcastMethod", "test");
+
+                var res = await ReadConnectionOutputAsync<InvocationDescriptor>(firstConnection.HttpConnection, serializer);
+                Assert.Equal("Broadcast", res.Method);
+                Assert.Equal(1, res.Arguments.Length);
+                Assert.Equal("test", res.Arguments[0]);
+
+                res = await ReadConnectionOutputAsync<InvocationDescriptor>(secondConnection.HttpConnection, serializer);
+                Assert.Equal("Broadcast", res.Method);
+                Assert.Equal(1, res.Arguments.Length);
+                Assert.Equal("test", res.Arguments[0]);
+
+                // kill the connections
+                firstConnection.Connection.Channel.Dispose();
+                secondConnection.Connection.Channel.Dispose();
+
+                await firstEndPointTask;
+                await secondEndPointTask;
+            }
+        }
+
+        [Fact]
+        public async Task HubsCanAddAndSendToGroup()
+        {
+            var serviceProvider = CreateServiceProvider();
+
+            var endPoint = serviceProvider.GetService<HubEndPoint<MethodHub>>();
+
+            using (var firstConnection = new ConnectionWrapper())
+            using (var secondConnection = new ConnectionWrapper())
+            {
+                var firstEndPointTask = endPoint.OnConnectedAsync(firstConnection.Connection);
+                var secondEndPointTask = endPoint.OnConnectedAsync(secondConnection.Connection);
+
+                await firstConnection.HttpConnection.Input.ReadingStarted;
+                await secondConnection.HttpConnection.Input.ReadingStarted;
+
+                var invocationAdapter = serviceProvider.GetService<InvocationAdapterRegistry>();
+                var writer = invocationAdapter.GetInvocationAdapter("json");
+                var serializer = new JsonSerializer();
+
+                await SendRequest_IgnoreWrite(firstConnection.HttpConnection, writer, "GroupSendMethod", "testGroup", "test");
+                // check that 'secondConnection' hasn't received the group send
+                Assert.False(secondConnection.HttpConnection.Output.ReadAsync().IsCompleted);
+
+                await SendRequest_IgnoreWrite(secondConnection.HttpConnection, writer, "GroupAddMethod", "testGroup");
+
+                await SendRequest_IgnoreWrite(firstConnection.HttpConnection, writer, "GroupSendMethod", "testGroup", "test");
+                // check that 'secondConnection' has received the group send
+                var res = await ReadConnectionOutputAsync<InvocationDescriptor>(secondConnection.HttpConnection, serializer);
```

Comment:
I think you should also check that the firstConnection did not receive the group message since it is not a group member